filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
 func NewFilter(pattern string) (Filter, error) {
-	idx := strings.Index(pattern, WildcardChar)
+	nIdx := strings.Index(pattern, negationChar)
+	if nIdx != 0 {
```

Comment:
Sanitize is only used to sanitize the filter expressions so that we can emit metrics on these expressions without special chars (ie the metric we emit to count the number of filter hits has a tag called expression and we want to sanitize the value of that tag). It's not used to sanitize incoming metric IDs.